messaging/src/test/java/org/axonframework/queryhandling/DefaultQueryBusSpanFactoryTest.java

```diff
@@ -0,0 +1,159 @@
+        test(builder -> builder,
+             spanFactory -> spanFactory.createScatterGatherSpan(queryMessage, true),
+             expectedSpan("QueryBus.scatterGatherQueryDistributed", TestSpanFactory.TestSpanType.DISPATCH)
+                     .withMessage(queryMessage)
+        );
+    }
+
+    @Test
+    void createsScatterGatherHandlerSpan() {
+        SubscriptionQueryMessage<?, ?, ?> queryMessage = Mockito.mock(SubscriptionQueryMessage.class);
+        test(builder -> builder,
+             spanFactory -> spanFactory.createScatterGatherHandlerSpan(queryMessage, 4),
+             expectedSpan("QueryBus.scatterGatherQuery-4", TestSpanFactory.TestSpanType.INTERNAL)
+                     .withMessage(queryMessage)
+        );
+    }
+
+    @Test
+    void createsStreamingQuerySpanNonDistributed() {
+        QueryMessage<?, ?> queryMessage = Mockito.mock(QueryMessage.class);
+        test(builder -> builder,
+             spanFactory -> spanFactory.createStreamingQuerySpan(queryMessage, false),
+             expectedSpan("QueryBus.streamingQuery", TestSpanFactory.TestSpanType.INTERNAL)
+                     .withMessage(queryMessage)
+        );
+    }
+
+    @Test
+    void createsStreamingQuerySpanDistributed() {
+        QueryMessage<?, ?> queryMessage = Mockito.mock(QueryMessage.class);
+        test(builder -> builder,
+             spanFactory -> spanFactory.createStreamingQuerySpan(queryMessage, true),
+             expectedSpan("QueryBus.streamingQueryDistributed", TestSpanFactory.TestSpanType.DISPATCH)
+                     .withMessage(queryMessage)
+        );
+    }
+
+    @Test
+    void createsQueryProcessingSpanDistributed() {
+        QueryMessage<?, ?> queryMessage = Mockito.mock(QueryMessage.class);
+        test(builder -> builder.distributedInSameTrace(true),
+             spanFactory -> spanFactory.createQueryProcessingSpan(queryMessage),
+             expectedSpan("QueryBus.processQueryMessage", TestSpanFactory.TestSpanType.HANDLER_CHILD)
+                     .withMessage(queryMessage)
+        );
+    }
+
+    @Test
+    void createsQueryProcessingSpanDistributedButSeparateTrace() {
+        QueryMessage<?, ?> queryMessage = Mockito.mock(QueryMessage.class);
+        test(builder -> builder.distributedInSameTrace(false),
+             spanFactory -> spanFactory.createQueryProcessingSpan(queryMessage),
+             expectedSpan("QueryBus.processQueryMessage", TestSpanFactory.TestSpanType.HANDLER_LINK)
+                     .withMessage(queryMessage)
+        );
+    }
+
+    @Test
+    void createsResponseProcessingSpan() {
+        QueryMessage<?, ?> queryMessage = Mockito.mock(QueryMessage.class);
+        test(builder -> builder,
+             spanFactory -> spanFactory.createResponseProcessingSpan(queryMessage),
+             expectedSpan("QueryBus.processQueryResult", TestSpanFactory.TestSpanType.INTERNAL)
+                     .withMessage(queryMessage)
+        );
+    }
+
```

Comment:
Curious: why not test the `propogateContext` operation?

Reply:
Test added